drivers/serial/Kconfig.nrfx

```diff
@@ -26,10 +26,17 @@ config UART_NRFX_UARTE
 	def_bool y
 	depends on DT_HAS_NORDIC_NRF_UARTE_ENABLED
 
+config UART_NRFX_UARTE_LEGACY_SHIM
+	depends on UART_NRFX_UARTE
+	# New shim takes more ROM. Until it is fixed use legacy shim in memory
+	# constraint case.
```

Comment:
is this expected to improve/be fixed?

Reply:
Yes, eventually we want to have one shim only. But harder to say exact when.